### PR TITLE
reformat codebase using python/black

### PR DESCRIPTION
**Pull request by dneise**
I open this PR just to see if the test are still running after reformatting using [`python/black`](https://github.com/python/black)

### CODE REVIEW

**Comment by dneise** [comment]
Since black does not exist for Py2, I first decided to verify codestyle only on Py3 (after all the style check is Py version independent). But having conditional steps turned out to be rather hard.

In the end I thought: 

Why not define one dedicated job for all style checks? 

So now the Style checks are run using py3.7 and for all other Pythons we skip the style checks and only build and run the test suite.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,3 +18,18 @@ Please submit patches by pull request where possible. Patches should add a test
...
+We use [python/black](https://github.com/python/black) in order to make our lives easier.
+We propose you do the same within this project, otherwise you might be asked to
+reformat your pull requests.
+
+It's really simple just:
+
+    pip install black
+    black .
+
+And there are plug ins for many editors.
```
Comment (cclauss) [inline CONTRIBUTING.md]: [plug ins for many editors](https://black.readthedocs.io/en/stable/editor_integration.html)

Comment (cclauss) [reply CONTRIBUTING.md]: Also "plugins" is a single word (no space).

Comment (dneise) [reply CONTRIBUTING.md]: Thanks - done

**Comment by dneise** [comment]
@cclauss  and @StoicLoofah could this be merged?

**Comment by StoicLoofah** [comment]
sorry for being delinquent on merging the other branches, but there are now conflicts to resolve here. @dneise can you take a look at those?

Regarding the discussion on documenting and enforcing black, this is exactly what i was hoping for. thanks for integrating those!

... now i need to get black setup myself =)

**Comment by cclauss** [comment]
The precommit hook is the way to go...  It blackens the code in the commit process.

**Comment by dneise** [comment]
@StoicLoofah ok, I think this can be merged now

**Comment by dneise** [comment]
@StoicLoofah 

I am sorry, this PR has a bit of an ugly commit history. Maybe you could to a [squash + merge](https://help.github.com/en/articles/about-pull-request-merges#squash-and-merge-your-pull-request-commits) commit? in order to have a cleaner history?

After all, the only thing I did in this PR was `black .` and then figure out how to verify blackness on circleci.

**Review by StoicLoofah — APPROVED** [review]
Looks great! Good suggestion on the squash.

**Pull request merged**